packages/clerk-js/src/ui/components/ApiKeys/ApiKeysTable.tsx

```diff
@@ -0,0 +1,189 @@
+import { useClerk } from '@clerk/shared/react';
+import type { ApiKeyResource } from '@clerk/types';
+import { useEffect, useState } from 'react';
+import useSWR from 'swr';
+
+import { timeAgo } from '../../../utils/date';
+import {
+  Button,
+  descriptors,
+  Flex,
+  Icon,
+  Input,
+  localizationKeys,
+  Spinner,
+  Table,
+  Tbody,
+  Td,
+  Text,
+  Th,
+  Thead,
+  Tr,
+} from '../../customizables';
+import { ThreeDotsMenu } from '../../elements';
+import { useClipboard } from '../../hooks';
+import { Check, Copy, Eye, EyeSlash } from '../../icons';
+
+const useApiKeySecret = ({ apiKeyID, enabled }: { apiKeyID: string; enabled: boolean }) => {
+  const clerk = useClerk();
+
+  return useSWR(enabled ? ['api-key-secret', apiKeyID] : null, () => clerk.getApiKeySecret(apiKeyID));
+};
+
+const CopySecretButton = ({ apiKeyID }: { apiKeyID: string }) => {
+  const [enabled, setEnabled] = useState(false);
+  const { data: apiKeySecret } = useApiKeySecret({ apiKeyID, enabled });
+  const { onCopy, hasCopied } = useClipboard(apiKeySecret ?? '');
+
+  useEffect(() => {
+    if (enabled && apiKeySecret) {
+      onCopy();
+      setEnabled(false);
+    }
+  }, [enabled, apiKeySecret, onCopy]);
+
+  return (
+    <Button
+      variant='ghost'
+      sx={{ margin: 1 }}
```

Comment:
Can this be handled by gap or alignment from the parent. Not sure about 1px margin usage here.

Reply:
Good catch! Removed. This is already handled in the parent which has a gap.